core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala

```diff
@@ -102,95 +106,101 @@ protected[actions] trait PrimitiveActions {
-     * either there is an error in the get, or the promise has completed because it timed out. The promise MUST
-     * complete in the caller to terminate the polling.
-     */
-    private def pollDbForResult(
-        docid: DocId,
+    /** Periodically polls the db to cover missing active acks. */
+    private def periodicallyPollForActivation(
+        user: Identity,
         activationId: ActivationId,
-        promise: Promise[WhiskActivation])(
-            implicit transid: TransactionId): Unit = {
-        // check if promise already completed due to timeout expiration (abort polling if so)
-        if (!promise.isCompleted) {
-            WhiskActivation.get(activationStore, docid) map {
-                activation => promise.trySuccess(activation.withoutLogs) // Logs always not provided on blocking call
-            } onFailure {
-                case e: NoDocumentException =>
-                    Thread.sleep(500)
-                    logging.debug(this, s"[POST] action activation not yet timed out, will poll for result")
-                    pollDbForResult(docid, activationId, promise)
-                case t: Throwable =>
-                    logging.error(this, s"[POST] action activation failed while waiting on result: ${t.getMessage}")
-                    promise.tryFailure(t)
-            }
-        } else {
-            logging.info(this, s"[POST] action activation timed out, terminated polling for result")
-        }
+        promise: Promise[Either[ActivationId, WhiskActivation]])(
+            implicit transid: TransactionId) = {
+
+        val datastorePollPeriodForActivation = 15.seconds
+        val docid = DocId(WhiskEntity.qualifiedName(user.namespace.toPath, activationId))
+
+        Scheduler.scheduleWaitAtMost(
+            datastorePollPeriodForActivation,
+            initialDelay = datastorePollPeriodForActivation,
+            name = "DbPoll",
+            logLevel = InfoLevel)(() => {
+                Option(promise) match {
+                    case Some(p) if !p.isCompleted =>
+                        WhiskActivation.get(activationStore, docid) map {
+                            // complete the future, which in turn will poison pill this scheduler
+                            activation => p.trySuccess(Right(activation.withoutLogs)) // Logs always not provided on blocking call
+                        } andThen {
+                            case Failure(e: NoDocumentException) => // do nothing, scheduler will reschedule another poll
+                            case Failure(t: Throwable) => // something went wrong, abort
+                                logging.error(this, s"failed while waiting on result: ${t.getMessage}")
+                                p.tryFailure(t) // complete the future, which in turn will poison pill this scheduler
+                        }
+
+                    case _ => throw new Exception("activation completed") // will halt the scheduler
```

Comment:
Why not send a `PoisonPill` to self? Is this also the reason why logLevel is set to Info?